docs(infrastructure): table of contents restructure

Michael-N-M opened the pull request:
This pull request restructures the table of contents and adds topics as per[ agreed document restructure format  ](https://docs.google.com/spreadsheets/d/1pQ6fqAOraA_Nt-i77bHz6x1Uaj2XXsLcuy0Dg9k6Cig/edit#gid=0)

Michael-N-M committed:
Created Infrastructure & database.md

Michael-N-M committed:
Created deploying.md

Michael-N-M committed:
Deleting Infrastructure Directory 

Sub-folder created in the wrong folder.

Michael-N-M committed:
infrastructure folder and deploying.md

Michael-N-M committed:
Create databases.md

Michael-N-M committed:
Rename docs/guide/docs/advanced/cluster.md to docs/guide/docs/infrastructure/cluster.md

Michael-N-M committed:
Rename docs/guide/docs/advanced/hosting.md to docs/guide/docs/infrastructure/hosting.md

Michael-N-M committed:
Rename docs/guide/docs/advanced/performances.md to docs/guide/docs/infrastructure/performances.md

Michael-N-M committed:
Rename docs/guide/docs/advanced/pipeline.md to docs/guide/docs/infrastructure/pipeline.md

Michael-N-M committed:
Rename docs/guide/docs/advanced/versions.md to docs/guide/docs/infrastructure/versions.md

Michael-N-M committed:
Rename docs/guide/docs/pro/rbac.md to docs/guide/docs/infrastructure/rbac.md

Michael-N-M committed:
Update sidebars.json

This update restructures the table of contents for infrastructure and deployment to conform to agreed changes as per documentation restructure agreement

marcmerce commented on docs/guide/website/sidebars.json:
```diff
@@ -1,8 +1,8 @@
     "Release Notes": ["releases/migrate", "updating"],
-    "Getting Started": ["introduction", "installation", "quickstart"],
-    "Main Concepts": [
+    "Getting Started": ["installation", "quickstart"],
```
Let's keep the introduction. I don't think the "what is botpress, why botpress, etc." has its place in installation or quickstart.

marcmerce commented on docs/guide/website/sidebars.json:
```diff
@@ -34,8 +34,18 @@
+      "infrastructure/hosting",
+      "infrastructure/performances",
+      "infrastructure/pipeline",
+      "infrastructure/rbac",
```
This should be placed within the bots and user management IMO. I know the GSheet says otherwise. @EFF What's your take on this?

EFF replied:
RBAC isn't infrastructure related indeed. We don't have a section for this but it would be "collaboration" so yes you can move this on bot building

EFF replied:
Also you have a pro/rbac section which is the same I guess ?

marcmerce replied:
@Michael-N-M Let's keep the RBAC where it is now, and when we are ready to make the Managing chatbots and users section, we'll transfer it there.

Commit:
Updated the advanced guide section

Removed from advanced guide pages who are now in deployment and infra. Moved the advanced section at the bottom.

marcmerce commented on docs/guide/website/sidebars.json:
```diff
@@ -13,17 +13,6 @@
       "main/nlu",
       "main/emulator"
     ],
-    "Advanced Guides": [
```
@Michael-N-M I have removed the pages that you renamed to place under deployment and infrastructures and I moved the rest at the bottom of the documentation. Please double check to make sure I didn't forget anything.

Michael-N-M committed:
Rename docs/guide/docs/infrastructure/rbac.md to docs/guide/docs/pro/rbac.md

Placed rbac back in the pro folder

Michael-N-M committed:
Update and rename docs/guide/docs/tutorials/database.md to docs/guide/docs/infrastructure/database.md

Moved database.md to the infrastructure folder.

Michael-N-M committed:
Update sidebars.json

Verified links and general structure.

Commit:
Merge branch 'master' into mm-docs-infrastructure-toc

Commit:
Merge branch 'master' into mm-docs-infrastructure-toc
